src/GNDStk/Node/src/detail.hpp

```diff
@@ -0,0 +1,45 @@
+
+// -----------------------------------------------------------------------------
+// apply_keyword
+// -----------------------------------------------------------------------------
+
+namespace detail {
+
+// apply_keyword<RESULT>
+template<class RESULT>
+class apply_keyword {
+public:
+   template<class NODE>
+   void operator()(const NODE &node) const
+   {
+      RESULT type{};
+      convert(node,type);
+   }
+};
+
+// apply_keyword<Node>
+template<
+   template<class...> class METADATA_CONTAINER,
+   template<class...> class CHILDREN_CONTAINER
+>
+class apply_keyword<Node<METADATA_CONTAINER,CHILDREN_CONTAINER>> {
```

Comment:
I don't understand what this (or the next one) is doing.

Reply:
This class relates to an extra nicety that's supported by my child_t type. I haven't described this feature to you yet; it would probably be worthwhile to have that discussion at some point. The intention was to lay some groundwork for the "dictionary" or "x-ray" capabilities Patrick spoke of a while back. I think you'll both like the feature; we just need to talk about it so that you understand what I'm doing.